tests/verify_dt_data_upgrade.erl

```diff
@@ -0,0 +1,180 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%% @doc r_t to verify changes to serialization/metadata/changes
+%%       across dt upgrades
+
+-module(verify_dt_data_upgrade).
+
+-export([confirm/0]).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-define(MAP_TYPE, <<"maps">>).
+-define(SET_TYPE, <<"sets">>).
+-define(SET_CAP, riak_dt_orswot).
+-define(BUCKET_M, {?MAP_TYPE, <<"testbucket">>}).
+-define(BUCKET_S, {?SET_TYPE, <<"testbucket">>}).
+-define(KEY, <<"flipit&reverseit">>).
+-define(CONFIG,
+        [
+         {riak_core,
+          [{ring_creation_size, 8},
+           {anti_entropy_build_limit, {100, 1000}},
+           {anti_entropy_concurrency, 8}]
+         }
+        ]).
+
+confirm() ->
+    TestMetaData = riak_test_runner:metadata(),
+    OldVsn = proplists:get_value(upgrade_version, TestMetaData, lts),
+
+    NumNodes = 4,
+    Vsns = [{OldVsn, ?CONFIG} || _ <- lists:seq(1, NumNodes)],
+
+    Nodes = [Node1,Node2|Nodes34] = rt:build_cluster(Vsns),
+
+    %% Create PB connection.
+    {P1, P2} = lists:split(1, Nodes),
+
+    Pid1 = rt:pbc(hd(P1)),
+    Pid2 = rt:pbc(hd(P2)),
+
+    rt:create_and_activate_bucket_type(Node1,
+                                       ?SET_TYPE,
+                                       [{datatype, set}]),
+    rt:create_and_activate_bucket_type(Node1,
+                                       ?MAP_TYPE,
+                                       [{datatype, map}]),
+
+    %% Partition the Cluster into 2 Sides
+    Partition = rt:partition(P1, P2),
+
+    try
```

Comment:
as @nickelization pointed out... the after still occurs if we crash out. Quick experiment to see if we need it anyhow then ;).